File: uservice-timeentries/host/src/main/java/sinnet/grpc/reports/report3/DtoDomainMapper.scala
package sinnet.report3

import sinnet.report3.grpc.{
  ReportRequest => ReportRequestDTO,
  CustomerDetails => CustomerDetailsDto,
  GroupDetails => GroupDetailsDto}
import java.time.LocalDate
import scala.collection.JavaConverters._
import scala.util.Try
import sinnet.reports._

/** Converts ReportRequest (DTO) to its local model. */
object DtoDomainMapper {

  implicit def toCustomerDetails(x: CustomerDetailsDto) = CustomerDetails(x.getName(), x.getAddress(), x.getCity());
  implicit def toGroupDetails(x: GroupDetailsDto) = GroupDetails(x.getPersonName(), x .getDetailsList.asScala.map(toCustomerDetails _));
    
  implicit def apply(dto: ReportRequestDTO): ReportRequest = {
    ReportRequest(dto.getDetailsList.asScala.map(toGroupDetails _))
  }
}

File: svc_timeentries/host/src/main/java/sinnet/grpc/reports/report3/DtoDomainMapper.java
package sinnet.grpc.reports.report3;

import sinnet.grpc.reports.report3.Models.CustomerDetails;
import sinnet.grpc.reports.report3.Models.GroupDetails;
import sinnet.grpc.reports.report3.Models.ReportRequest;

/** Converts ReportRequest (DTO) to its local model. */
interface DtoDomainMapper {

  static CustomerDetails toCustomerDetails(sinnet.report3.grpc.CustomerDetails x) { 
    return new CustomerDetails(x.getName(), x.getAddress(), x.getCity());
  }
  
  static GroupDetails toGroupDetails(sinnet.report3.grpc.GroupDetails x) {
    return new GroupDetails(x.getPersonName(), x.getDetailsList().stream().map(it -> toCustomerDetails(it)).toList());
  }
    
  static ReportRequest apply(sinnet.report3.grpc.ReportRequest x) {
    return new ReportRequest(x.getDetailsList().stream().map(it -> toGroupDetails(it)).toList());
  }
}
